source/platform/file_formats/ncch.hpp

```diff
@@ -38,7 +38,37 @@ using MediaUnit32 = MediaUnit<uint32_t>;
 
 using Bytes32 = uint32_t;
 
+
 struct NCCHHeader {
+    struct TypeFlags {
+        enum ContentType {
+            Unspecified          = 0,
+            SystemUpdate         = 1,
+            InstructionManual    = 2,
+            DLPChild             = 3,
+            Trial                = 4,
+            ExtendedSystemUpdate = 5,
+        };
+        
+        enum FormType {
+            NotAssigned            = 0,
+            SimpleContent          = 1,
+            ExecutableWithoutRomFS = 2,
+            Executable             = 3,
+        };
+        
+        BOOST_HANA_DEFINE_STRUCT(TypeFlags,
+            (uint8_t, raw)
+        );
+        
+        constexpr auto form_type() { return BitField::v3::MakeFieldOn<0, 2>(this); }
+        constexpr auto content_type() { return BitField::v3::MakeFieldOn<2, 6>(this); }
```

Comment:
BitField v3 supports an exposure type as the third parameter:
```suggestion
        constexpr auto form_type() { return BitField::v3::MakeFieldOn<0, 2, FormType>(this); }
        constexpr auto content_type() { return BitField::v3::MakeFieldOn<2, 6, ContentType>(this); }
```